src/lib/mina_net2/libp2p_helper.ml

```diff
@@ -0,0 +1,288 @@
+open Async
+open Core
+open Pipe_lib
+open O1trace
+
+exception Libp2p_helper_died_unexpectedly
+
+module Go_log = struct
+  let ours_of_go lvl =
+    let open Logger.Level in
+    match lvl with
+    | "error" | "panic" | "fatal" ->
+        Error
+    | "warn" ->
+        Debug
+    | "info" ->
+        (* this is intentionally debug, because the go info logs are too verbose for our info *)
```

Comment:
TODO: create an issue to modify logging config of the Go process not to label `warn` and `info` as `debug`.